compiler/rustc_session/src/options.rs

```diff
@@ -999,8 +999,8 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
     mir_emit_retag: bool = (false, parse_bool, [TRACKED],
         "emit Retagging MIR statements, interpreted e.g., by miri; implies -Zmir-opt-level=0 \
         (default: no)"),
-    mir_opt_level: usize = (1, parse_uint, [TRACKED],
-        "MIR optimization level (0-3; default: 1)"),
+    mir_opt_level: Option<usize> = (None, parse_opt_uint, [TRACKED],
+        "MIR optimization level (0-3; default: Some(1) in non optimized builds and Some(2) in optimized builds)"),
```

Comment:
Should it be 0-4 here, and probably not mention the Some() since it's user-facing text :slightly_smiling_face: ?